typescript-test-samples/typescript-sample-app/test/helpers/repeater.ts

```diff
@@ -0,0 +1,29 @@
+type RepeaterConfig = {
+  interval: number; // milliseconds
+  limit: number;
+};
+
+type RepeatResult<T> = {
+  result: T | null | undefined;
+  count: number;
+};
+
+export class Repeater {
+  private readonly interval: number;
+  private readonly limit: number;
+  private count = 0;
+
+  constructor({ interval, limit }: RepeaterConfig) {
+    this.interval = interval;
+    this.limit = limit;
```

Comment:
Nice job putting a limit in and not looping forever assuming it will eventually work.